packages/dashboard/src/layout/usePanelRegistration.ts

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import {
+  DashboardPluginComponentProps,
+  PanelComponentType,
+  PanelDehydrateFunction,
+  PanelHydrateFunction,
+  PanelProps,
+} from '../DashboardPlugin';
+
+/**
+ * Registers a given panel component. Also runs a `useEffect` that will
+ * automatically de-register then panel on unmount.
+ * @param registerComponent
```

Comment:
We might not want this as part of the hook. Bender and I chatted about plugins the other day and there's a lot of rough edges and things that need to be cleaned up around their implementation. Basically we _should_ be able to write our existing plugins all as external plugins in an ideal world

hydrate and dehydrate are the same. IMO they are logic that belong in the component, not at the registration point. Our existing hydration code is in AppMainContainer which separates it from where the hydration is actually used.

I don't think they're changes that should be in this PR, but just something to note that this will probably change signatures as we address those points